packages/block-editor/src/components/button-block-appender/index.js

```diff
@@ -67,6 +67,7 @@ function ButtonBlockAppender(
 							<VisuallyHidden as="span">{ label }</VisuallyHidden>
 						) }
 						<Icon icon={ plus } />
+						{ children }
```

Comment:
This may need to be a separate PR. Basically it allows us to do this

```
<ButtonBlockAppender
	className="offcanvas-editor__list-appender__toggle"
	rootClientId={ rootClientId }
>
	{ __( 'Add Menu Item' ) }
</ButtonBlockAppender>
```